### PR TITLE
[DPP-75] Explicitly deflate/inflate data outside of the index

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
[Integration Kit] Compression and decompression of stored DAML-LF values
is now executed outside of the index database, allowing to make more
efficient use of the participant resources when indexing.
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
This is currently a draft while waiting for #8019 to be merged, but it's ready for review.

File: ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      val output = new ByteArrayOutputStream(uncompressed.length)
+      val gzip = compress(output)
+      try {
+        gzip.write(uncompressed)
+      } finally {
+        gzip.close()
+      }
+      val compressed = output.toByteArray
+      output.close()
+      ratio.update(compressed.length * 100 / uncompressed.length)
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala:24]: I'd turn the ratio around

Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala:24]: Ultimately I'll probably do what you originally did and simply record compressed and uncompressed size separately, but thanks.

Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala]: Addressed by https://github.com/digital-asset/daml/pull/8039/commits/1165b8bf5e604f18427f3998eb566ef098696966

**Comment by SamirTalwar-DA** [comment]
This PR description makes it sound like the compression behavior is being moved from the DB critical path to elsewhere, but from what I understand, it's actually _introducing_ compression, and making sure not to let it touch the DB critical path. Do I have that right?

**Comment by stefanobaghino-da** [comment]
> This PR description makes it sound like the compression behavior is being moved from the DB critical path to elsewhere, but from what I understand, it's actually _introducing_ compression, and making sure not to let it touch the DB critical path. Do I have that right?

Compression was already performed by the DBMS itself, this moves it outside of it.

**Comment by SamirTalwar-DA** [comment]
OK, cool, thanks for the clarification!

Do you know what the performance improvement looks like?

**Comment by stefanobaghino-da** [comment]
> OK, cool, thanks for the clarification!
> 
> Do you know what the performance improvement looks like?

@tudor-da Got somewhere around 10% improvement when working on [his PoC](https://github.com/digital-asset/daml/compare/DPP-52-compress-daml-lf-values) in indexing performance (measured by the integrity checker).

**Comment by stefanobaghino-da** [comment]
> > OK, cool, thanks for the clarification!
> > Do you know what the performance improvement looks like?
> 
> @tudor-da Got somewhere around 10% improvement when working on [his PoC](https://github.com/digital-asset/daml/compare/DPP-52-compress-daml-lf-values) in indexing performance (measured by the integrity checker).

I forgot to add that the problem right now is that instead _this_ PR causes a 10% _drop_ in performance. We are investigating why. For some reason the compression ratio is way better in @tudor-da's case and we don't (yet) understand why (especially since the code is almost the same, when it comes to compression).

**Comment by stefanobaghino-da** [comment]
After a lot of measurement, it's still unclear whether this would actually be adding any value, so we're dropping this.

**Pull request closed without merging**